src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +392,77 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
-      this.checkActionEligibility_(action.type)
-    );
-
-    // No audience actions means use the default prompt.
-    if (potentialActions.length === 0) {
-      return undefined;
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    const fetchedLocalStorage = /** @type {LocalStorage} */ ({});
+    let promiseChain = Promise.resolve();
+    const actionsIncludeSurvey = audienceActions
+      .map((action) => action.type)
+      .includes(TYPE_REWARDED_SURVEY);
+    if (actionsIncludeSurvey) {
+      promiseChain = promiseChain
+        .then(() =>
+          this.getEvent_(
+            completedActionToStorageKey_(
+              AnalyticsEvent.ACTION_SURVEY_DATA_TRANSFER
+            )
+          )
+        )
+        .then((surveyCompleted) => {
+          fetchedLocalStorage[STORAGE_KEY_SURVEY_COMPLETED] = surveyCompleted;
+        });
     }
 
-    // Default to the first recommended action.
-    let actionToUse = potentialActions[0].type;
-
-    // Contribution prompts should appear before recommended actions, so we'll need
-    // to check if we have shown it before.
-    if (
-      autoPromptType === AutoPromptType.CONTRIBUTION ||
-      autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
-    ) {
-      if (!dismissedPrompts) {
-        this.promptDisplayed_ = AutoPromptType.CONTRIBUTION;
-        return undefined;
-      }
-      const previousPrompts = dismissedPrompts.split(',');
-      potentialActions = potentialActions.filter(
-        (action) => !previousPrompts.includes(action.type)
+    return promiseChain.then(() => {
+      // No audience actions means use the default prompt.
+      let potentialActions = audienceActions.filter((action) =>
+        this.checkActionEligibility_(action.type, fetchedLocalStorage)
```

Comment:
Can we directly pass in `surveyCompleted`? (Would `surveyCompletions` be representing the value better which is timestamps of completions?)

Reply:
Sure, I originally implemented this way because of the condition around the survey action, but if that is removed then I can pass this directly.